api/project.rb

```diff
@@ -28,6 +28,7 @@
 end
 
 command = ARGV.first
-args = ARGV.drop(1)
 
-common.handle_or_die(command, *args)
+args = Hash[ ARGV.drop(1).flat_map{|s| s.scan(/--?([^=\s]+)(?:=(\S+))?/) } ]
+
+common.handle_or_die(command, args)
```

Comment:
Might consider passing the args array and using ruby's OptionParser to parse within a given command (but @dmohs will have a better perspective on this than me).